### ISSUE TITLE
Invalid E1019 error with nested Transform function

### ISSUE THREAD

**Issue by szechyjs** [aws-cloudformation/cfn-lint#3210]
### CloudFormation Lint Version

0.87.1

### What operating system are you using?

Ubuntu

### Describe the bug

Receive the following error `E1019 Sub parameter should use a valid function for Resources/FooS3Bucket/Properties/BucketName/Fn::Sub` however the following template is valid and builds just fine in CF.

### Expected behavior

It should not be an error. I suspect `Fn::Transform` needs added to the list of allowable functions here: https://github.com/aws-cloudformation/cfn-lint/blob/main/src/cfnlint/rules/functions/Sub.py#L32

### Reproduction template

```yaml
BucketName: !Sub
  - "${EnvironmentLower}-foo-bar"
  - EnvironmentLower:
      Fn::Transform:
        Name: String
        Parameters:
          InputString: !Ref Environment
          Operation: Lower
```

**Comment by kddejong** [comment]
We may be able to create a workaround for this one but in general we are tracking transform issues here #476

**Comment by kddejong** [comment]
Fn Transform in a sub shouldn't toss an error anymore.
